tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -530,3 +531,57 @@ def get_service_locations(cls, service):
         """
         return [location for location in GcpLocation._locations
                 if service in GcpLocation._locations[location]]
+
+
+class GCPIamPolicyFilter(ValueFilter):
```

Comment:
its unclear why this wouldn't be in the filters package vs in resource manager base class's module. 

also the implementation is a bit odd, in that its hardcoding the resource types, ideally this would be a subclass responsibility to fill in type specific metadata.

the use of executor factory is also odd since its hardcoded to a single thread, just using the main thread seems like it simplifies.

Reply:
this filter feels like it should be split out into a separate pr, there's no tests for it as well.